common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -1288,17 +1288,18 @@ def get_submission_metadata(self, answers, correct_map):
             if is_correct is None:
                 is_correct = ''
 
+            response_data = getattr(answer_input, 'response_data', {})
             input_metadata[input_id] = {
-                'question': answer_input.response_data.get('label', ''),
+                'question': response_data.get('label', ''),
                 'answer': user_visible_answer,
                 'response_type': getattr(getattr(answer_response, 'xml', None), 'tag', ''),
                 'input_type': getattr(answer_input, 'tag', ''),
                 'correct': is_correct,
                 'variant': variant,
             }
             # Add group_label in event data only if the responsetype contains multiple inputtypes
-            if answer_input.response_data.get('group_label'):
-                input_metadata[input_id]['group_label'] = answer_input.response_data.get('group_label')
+            if response_data.get('group_label'):
```

Comment:
@muhammad-ammar i'm still leaning toward removing this if condition and setting the default value for group_label with rest of the data.